src/parser/Expression.ts

```diff
@@ -216,7 +216,7 @@ export class FunctionExpression extends Expression implements TypedefProvider {
                 new SourceNode(this.asToken.range.start.line + 1, this.asToken.range.start.character, state.pathAbsolute, 'as'),
                 ' ',
                 //return type
-                new SourceNode(this.returnTypeToken.range.start.line + 1, this.returnTypeToken.range.start.character, state.pathAbsolute, this.returnTypeToken.text.toLowerCase())
+                util.bscTypeToTypeString(this.returnType)
```

Comment:
We still need to create a `SourceNode` here. The `SourceNode` tells the sourcemap generator that "here's where the original token was, and here's what the new text is". This line should look something like this:

```
new SourceNode(
    this.returnTypeToken.range.start.line + 1, 
    this.returnTypeToken.range.start.character, 
    state.pathAbsolute, 
    util.bscTypeToTypeString(this.returnType)
)
```